bw2data/utils.py

```diff
@@ -297,33 +296,30 @@ def download_file(filename, directory="downloads", url=None):
 
     assert isinstance(directory, str), "`directory` must be a string"
     dirpath = projects.request_directory(directory)
-    filepath = dirpath / filename
+    filepath = os.path.join(dirpath, filename)
     download_path = (url if url is not None else DOWNLOAD_URL) + filename
-    request = requests.get(download_path, stream=True)
-    if request.status_code != 200:
-        raise NotFound(
-            "URL {} returns status code {}.".format(download_path, request.status_code)
-        )
-    download = request.raw
-    chunk = 128 * 1024
-    with open(filepath, "wb") as f:
+    with urllib.request.urlopen(download_path) as response, open(filepath, 'wb') as out_file:
+        if response.status != 200:
+            raise NotFound(
+                "URL {} returns status code {}.".format(download_path, response.status)
+            )
+        chunk = 128 * 1024
         while True:
-            segment = download.read(chunk)
+            segment = response.read(chunk)
             if not segment:
                 break
-            f.write(segment)
+            out_file.write(segment)
     return filepath
 
 
 def web_ui_accessible():
```

Comment:
Please delete this function and `open_activity_in_webbrowser`. They are ancient an nonfunctional.